plugins/destination/managed_writer.go

```diff
@@ -71,46 +69,20 @@ func (p *Plugin) flush(ctx context.Context, metrics *Metrics, table *schema.Tabl
 	}
 }
 
-func (p *Plugin) removeDuplicatesByPK(table *schema.Table, resources [][]any) [][]any {
+func removeDuplicatesByPK(table *schema.Table, resources [][]any) [][]any {
 	// special case where there's no PK at all
 	if len(table.PrimaryKeys()) == 0 {
 		return resources
 	}
 
 	pks := make(map[string]struct{}, len(resources))
 	res := make([][]any, 0, len(resources))
-	var reported bool
 	for _, r := range resources {
 		key := pk.String(table, r)
-		_, ok := pks[key]
-		switch {
-		case !ok:
+		if _, ok := pks[key]; !ok {
 			pks[key] = struct{}{}
 			res = append(res, r)
-			continue
-		case reported:
-			continue
 		}
-
-		reported = true
-		pkSpec := "(" + strings.Join(table.PrimaryKeys(), ",") + ")"
-
-		// log err
-		p.logger.Error().
-			Str("table", table.Name).
-			Str("pk", pkSpec).
-			Str("value", key).
-			Msg("duplicate primary key")
```

Comment:
I understand and am for getting rid of the Sentry call below, but why get rid of the log message? Don't we consider that part of our public API?

Reply:
nice catch